linear_algebra/code/CMakeLists.txt

```diff
@@ -48,30 +48,15 @@ include_directories(include test .)
 
 add_executable(la_test ${Sources} ${Includes})
 
-set(CMAKE_VERBOSE_MAKEFILE 1)
-
-if(CXX_COMPILER STREQUAL clang++)
-    set(CMAKE_C_COMPILER   clang)
-    set(CMAKE_CXX_COMPILER clang++)
-else()
-    set(CMAKE_C_COMPILER   gcc)
-    set(CMAKE_CXX_COMPILER g++)
-endif()
-
-if(CMAKE_CXX_COMPILER STREQUAL clang++)
-
-    set(CMAKE_C_FLAGS_DEBUG   "-g -Wall -pedantic -Wextra")
-    set(CMAKE_CXX_FLAGS_DEBUG "-g -std=c++17 -Wall -pedantic -Wextra -Wno-unused-parameter -Wno-unused-function")
-
-    set(CMAKE_C_FLAGS_RELEASE   "-O3 -march=westmere -Wall -pedantic -Wextra")
-    set(CMAKE_CXX_FLAGS_RELEASE "-O3 -march=westmere -std=c++17 -stdlib=libc++ -Wall -pedantic -Wextra -Wno-unused-parameter -Wno-unused-function")
-
-elseif(CMAKE_CXX_COMPILER STREQUAL g++)
-
-    set(CMAKE_C_FLAGS_DEBUG   "-g -Wall -pedantic")
-    set(CMAKE_CXX_FLAGS_DEBUG "-g -std=c++17 -Wall -pedantic -Wextra  -Wno-unused-function -Wno-unused-variable -W -Wno-unused-but-set-variable -fmax-errors=10")
```

Comment:
We setting C++ 17 compiler flags via the target properties which will do the right thing for every compiler.  No need to set the flags for compilers directly.

Reply:
One possible problem.  As I understand it, setting the CXX_STANDARD as 17 enables -std=g++17 which is slightly different than -std=c++17.

Reply:
"std=g++17" is I believe G++ with compiler-specific extensions enabled?  However, I have disabled compiler extensions explicitly with the flag:
```
    CXX_STANDARD_REQUIRED YES
    CXX_EXTENSIONS NO
```
This article describes this technique in details: https://crascit.com/2015/03/28/enabling-cxx11-in-cmake/.  Hopefully, this should address your concerns.